### DISCUSSION TITLE
To create tables programmatically and tenant based config

### DISCUSSION THREAD

**Discussion by braj065** [serverpod/serverpod#803] [Q&A] [answered]
Hi,

1) Is it possible to create tables programmatically. Asking for the usecase where i want to onboard a new tenant, and i want to create all the associated tables from the code. So that i can prepend each table name with tenant name. For ex- tenantname_my_table_name.
2) Is it possible to have tenant based config. For ex- for different tenants the aws instance size and config would be different.
3) Also is it possible to use firebase phone no. and otp auth to use along with spod auth. Firebase phone-otp auth doesn't needs email of the user to be registered. On login with phone no.-otp it creates an uuid of the user in firebase auth. OTP comes through sms that firebase/google takes care of.

**Comment by vlidholt** [comment, marked as answer]
1. You can create new tables on the fly using custom SQL queries. However, I would probably be a much smarter idea to just assign an id to each new tenant and have a column with that id in your tables.
2. Not out of the box. Is there a reason for not running them on the same instances, if you are using the same database?
3. Yes, this should work.

I think perhaps a solution that could work for you is to have a separate deployment for each tenant. Then, they can have completely different databases, instance types, or configurations.

**Reply by braj065** [reply]
Thanks @vlidholt 
2) The multi tenant feature allows tenant based platform features. Each tenant can have a different authentication  method, separate billing for each tenant etc. I was referring to following from gcp - https://cloud.google.com/identity-platform/docs/multi-tenancy. I think it uses the same database(correct me if i am wrong) but the tenant based features provide silos for each tenant.
